python/ray/autoscaler/_private/autoscaler.py

```diff
@@ -241,20 +249,33 @@ def _update(self):
             return
 
         self.last_update_time = now
-        self.update_worker_list()
+
+        # Query node provider to update workers, all_workers, and all_nodes.
```

Comment:
imagine I am someone reading this code for the first time, this comment will make me wonder what the difference between workers, all_workers, all_nodes.

Reply:
will go with sasha's suggestion of

workers
unmanaged_nodes
head

Reply:
refactored